datumaro/components/visualizer.py

```diff
@@ -118,22 +150,122 @@ def vis_one_sample(
             else None
         )
 
+        context = defaultdict(list)
         for ann in annotations:
             if ann.type in self.ignored_types:
                 warnings.warn(f"{ann.type} in self.ignored_types. Skip it.")
                 continue
 
-            if ann.type == AnnotationType.bbox:
-                self._draw_bbox(ann, label_categories, ax)
+            if ann.type in self._draw_func:
+                self._draw_func[ann.type](ann, label_categories, fig, ax, context[ann.type])
             else:
-                raise NotImplementedError(f"{ann.type} is not implemented yet.")
+                raise ValueError(f"Unknown ann.type={ann.type}")
 
         ax.set_title(f"ID: {id}, Subset={subset}")
         ax.set_axis_off()
 
         return fig
 
-    def _draw_bbox(self, ann: Bbox, label_categories: Optional[LabelCategories], ax: Axes):
+    def _draw_label(
+        self,
+        ann: Label,
+        label_categories: Optional[LabelCategories],
+        fig: Figure,
+        ax: Axes,
+        context: List,
+    ) -> None:
```

Comment:
could you add some descriptions for each argument?

Reply:
In the next PR, https://github.com/openvinotoolkit/datumaro/pull/747/files#diff-d580edf6ed5ded10d766c687ad3717992364d871cba8271942e63a1f5f058de3R101-R108, it will be refactored into `_draw()`. I'll add descriptions in that PR because we have to add it to every `_draw_xxx()` function in this PR.